pkg/ccl/changefeedccl/changefeed_test.go

```diff
@@ -1172,6 +1177,59 @@ func TestChangefeedAuthorization(t *testing.T) {
 	t.Run(`enterprise`, enterpriseTest(testFn))
 }
 
+func TestChangefeedFailOnRBRChange(t *testing.T) {
+	defer leaktest.AfterTest(t)()
+	defer log.Scope(t).Close(t)
+
+	skip.UnderRace(t)
+	skip.UnderShort(t)
+	rbrErrorRegex := regexp.MustCompile(`CHANGEFEED cannot target REGIONAL BY ROW tables: rbr`)
+	assertRBRError := func(ctx context.Context, f cdctest.TestFeed) {
+		ctx, cancel := context.WithTimeout(ctx, 30*time.Second)
+		defer cancel()
+		done := make(chan struct{})
+		go func() {
+			if _, err := f.Next(); err != nil {
+				assert.Regexp(t, rbrErrorRegex, err)
+				done <- struct{}{}
+			}
+		}()
+		select {
+		case <-ctx.Done():
+			t.Fatal("timed out waiting for changefeed to fail")
+		case <-done:
+		}
+	}
+	testFn := func(t *testing.T, db *gosql.DB, f cdctest.TestFeedFactory) {
+		sqlDB := sqlutils.MakeSQLRunner(db)
+		sqlDB.Exec(t, "SET CLUSTER SETTING changefeed.experimental_poll_interval = '200ms'")
+		sqlDB.Exec(t, "SET CLUSTER SETTING kv.closed_timestamp.target_duration = '50ms'")
+		sqlDB.Exec(t, "SET CLUSTER SETTING kv.closed_timestamp.close_fraction = .99")
```

Comment:
I don't think these 3 are needed, are they?  I thought sinkless+enterprise test below do that already.

Reply:
I removed 2 of the 3.  The lower target_duration does seem to speed up this test substantially and right now we have a default of 1s.